### ISSUE TITLE
Way of organising channels

### ISSUE THREAD

**Issue by matthias-brun** [RocketChat/Rocket.Chat#2049]
Currently there is no way of sorting or grouping the channels a user is in, except for completely hiding them. Over time, as the number of channels a user is in grows, the channel list gets messy and it becomes difficult to quickly find a specific channel.

My suggestion would be to add a feature which allows for
* either organising the channels in __directories or groups__ (every user can have a different layout and is able to organise in a way that fits best for them)
* or the creation of __subchannels__ which would allow the same functionality but globally, so all users have the same structure.

The exact implementation could look quite different than what I outlined above but should provide a way of organising many channels.

I couldn't find any similar issues, has this never been discussed before?

**Comment by lunitic** [comment]
@engelgabriel Has any work been done regarding grouping of channels?

**Comment by lunitic** [comment]
Have there been any progress?

**Comment by mixrat** [comment]
+1 for subgroups, i.e....
```
Rooms(Groups)
       |__Channel 1
       |__Channel 2
       |__Channel 3

Direct 1:1 Chats
```

**Comment by localguru** [comment]
At this point I like the team concept of mattermost:
https://docs.mattermost.com/help/getting-started/creating-teams.html

**Comment by simon-doerflinger** [comment]
Are there any news to this topic? It would be a great Feature and could help us a lot 😁

**Comment by stgolem** [comment]
We also need Teams. Also there has to be Team Administrators, that have more permissions inside Team channels.

**Comment by Atomy9** [comment]
Yes this feature would be highly appreciated, any news on the progress ?

**Comment by phischdev** [comment]
This is the last essential feature that prevents us from entirely switching to rocket.chat

**Comment by kenton-r** [comment]
What about groupings in the side bar by taxonomy tags. It could be listed only under the first tags or listed multiple times under each tag.

**Comment by neokeld** [comment]
Similar issue : #1299

**Comment by adityanmishra** [comment]
This will be a nice feature. Currently we are using rocket chat for all our notifications for many teams. It will really be helpful if we can group the channels

**Comment by jaenulton** [comment]
Collapsible would be a must. The feature request is nearly 4 years old, any update yet?

**Comment by neylwalecki** [comment]
+1

**Comment by Balecan** [comment]
I agree with previous post. This feature is very interesting to organize our teams channel. Some ppl know equivalent feature in other tools and it could be to find the same thing in Rocket Chat to help improve is usage. Ty

**Comment by clemoan** [comment]
+1

**Comment by BrennenHodge** [comment]
This feature is even more important with the increase of remote teams. 

I vote we bump this up to higher priority.

**Comment by nemeth-it** [comment]
+1

**Comment by avatar1024** [comment]
+1 for this too! It would be very useful. It also feels that we should be able to sort discussions as sub channels (since they are created within a channel). So it seems like a three level tree would be best:
Room: top level which is just used to group channels > can give it a description but not for people to post messages
Channels: top layer for people to chat
Discussions: specific topic within a channel > effectively sub channels

**Comment by charliehelmijr** [comment]
> +1 for this too! It would be very useful. It also feels that we should be able to sort discussions as sub channels (since they are created within a channel). So it seems like a three level tree would be best:
> Room: top level which is just used to group channels > can give it a description but not for people to post messages
> Channels: top layer for people to chat
> Discussions: specific topic within a channel > effectively sub channels

This would be perfect.

**Comment by Forrestus** [comment]
+1

**Comment by GrzegorzBobrowicz** [comment]
+1

**Comment by peiacademics** [comment]
+100 for this feature. Its really useful and would make things so much more efficient. 

Check : Telegram's folder structure also. Everything is moving towards better organization.

**Comment by tetefolle** [comment]
+1

**Comment by Tejeev** [comment]
+1
Easily the most desirable improvement to RocketChat for me

**Comment by Creaous** [comment]
+1

**Comment by Verfuchst** [comment]
+1

**Comment by robbyoconnor** [comment]
THis would be cool -- I could split out for example Google Summer of Code by year!

**Comment by samdup** [comment]
+1 - lack of channel organization makes the communication tool get more and more cumbersome and intimidating with time. I'd love to see this feature (ideally for admins) with Rocket.chat! 💖

Here's what some competitors have done
- [Slack has this as a "sections" feature](https://slack.com/help/articles/360043207674-Organize-your-sidebar-with-custom-sections#organize-your-conversations). On enterprise-grid you can control the sections for all users.
- [Discord has categories and subchannels](https://support.discord.com/hc/en-us/articles/115001580171-Channel-Categories-101) that admins can control
- [Mattermost enables users to organize channels](https://community-daily.mattermost.com/core/channels/ux-channel-sidebar), but the admin can't organize it for all users.

**Comment by hawatius** [comment]
+1

**Comment by ankar84** [comment]
Other issues related to that theme
https://github.com/RocketChat/Rocket.Chat/issues/12119
https://github.com/RocketChat/Rocket.Chat/issues/18803
https://github.com/RocketChat/feature-requests/issues/451

**Comment by v360-eduardo-marques** [comment]
+1

**Comment by aakash-gitdev** [comment]
I think this issue still persists. The list in the channel section should be made collapsible. Are there any updates regarding this?

**Comment by vesperk** [comment]
+1

**Comment by Meibisu** [comment]
+1

**Comment by adestis-ds** [comment]
I vote for including this into the mid-term milestone!

**Comment by clevertree** [comment]
If it's not obvious yet why this feature was held back for 4 years, when it would take an afternoon to complete - 
**Corporate Sabotage**. 
Big Tech can't compete with open source, but they can certainly pay off free techs of open source projects like this one to sabotage their own work, or in this particular case, make the product less competitive with BigTech. I don't think anything short of naming and shaming will slow the spread of **Corporate Sabotage** in open source projects >:(

**Comment by clevertree** [comment]
Sure, most of the time, but it's 2022, and it appears that planned obsolescence has hit the open source industry, and hard. I'd imagine if these guys aren't paid off to slow the release of industry-competitive product features, that they represent the exception to the rule in 2022

**Comment by subspecs** [comment]
> > This feature is required since 2016, and still no reply from developers...
> 
> @Meibisu Why not implement it yourself? This is open-source, don't demand things.

Well we're sorry that we asked such a basic necessity of the poor poor developers.

**Comment by Meibisu** [comment]
> > This feature is required since 2016, and still no reply from developers...
> 
> @Meibisu Why not implement it yourself? This is open-source, don't demand things.

Because i use their enterprise license. And dont want to implement things and change them with every update. That's developers task and very important improvement which available in many others messendgers.

**Comment by jimintz** [comment]
Obligatory 2023 bump
